pkg/util/cloudfoundry/cloudfoundry.go

```diff
@@ -14,18 +14,37 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util"
 )
 
-// GetHostAlias returns the host alias from Cloud Foundry
-func GetHostAlias() (string, error) {
+// Define alias in order to mock in the tests
+var getFqdn = util.Fqdn
+
+// GetHostAliases returns the host aliases from Cloud Foundry
+func GetHostAliases() ([]string, error) {
 	if !config.Datadog.GetBool("cloud_foundry") {
 		log.Debugf("cloud_foundry is not enabled in the conf: no cloudfoudry host alias")
-		return "", nil
+		return nil, nil
 	}
 
+	aliases := []string{}
+
+	// Always send the bosh_id if specified
 	boshID := config.Datadog.GetString("bosh_id")
 	if boshID != "" {
-		return boshID, nil
+		aliases = append(aliases, boshID)
 	}
 
 	hostname, _ := os.Hostname()
-	return util.Fqdn(hostname), nil
+	fqdn := getFqdn(hostname)
+
+	if config.Datadog.GetBool("cf_os_hostname_aliasing") {
```

Comment:
Why don't we do this all the time?

Reply:
I don't want to potentially break current configs of users